src/peft/tuners/adaption_prompt/layer.py

```diff
@@ -100,6 +104,15 @@ def forward(self, **kwargs):
         query_states = compute_query_states(model=self.model, **kwargs)
 
         previous_dtype = query_states.dtype
+
+        # Reshape and average the extra tensors
+        query_states_reshaped = query_states.reshape(
+            bsz, self.model.num_heads, -1, (self.model.head_dim // factor), factor
+        )
+
+        # Take the mean along the last dimension to get [bsz, 32, X, 32]
```

Comment:
Just for my understanding, what is the intuition behind taking the mean here?

Reply:
The shape of query states is not compatible of that of adapter_k, which is required to calculate the adapter_output. Hence, to reshape it I took the average. Another way would be to drop the extra states, however averaging can prevent loss of information(to some extent). As a result I took the average. 
Please let me know if there is a better way!